app/models/region.rb

```diff
@@ -1,20 +1,25 @@
 class Region < ApplicationRecord
+  MAX_LABEL_LENGTH = 255
+
   ltree :path
   extend FriendlyId
   friendly_id :name, use: :slugged
 
+  attr_writer :parent
+
+  def parent
+    @parent || super
+  end
+
   validates :name, presence: true
   validates :slug, presence: true, uniqueness: true
   validates :path, presence: true, uniqueness: true
   validates :region_type, presence: true
 
   belongs_to :source, polymorphic: true, optional: true
 
-  before_discard do
-    self.path = nil
-  end
-
-  MAX_LABEL_LENGTH = 255
+  before_validation :set_path
```

Comment:
I don't think we want to set the path every time a region is validated -- the path should be stable and unchanging most of the time, since the slug itself and the hiearchy of regions is unchanging most of the time.

Reply:
Pushing up some changes to handle this concern and limit the scope of when we are resetting a path, I'll comment on the code once I'm sure all tests are passing.